docs/distributed_scenarios.rst

```diff
@@ -0,0 +1,332 @@
+Running distributed scenarios
+=============================
+
+Mahiru is a distributed system. While that doesn't stop anyone from running a
+complete Mahiru system on a single machine, some experiments require setting up
+Mahiru on actual different machines to get realistic results.
+
+So, we need some way of running distributed experiments: we obtain some
+(physical or virtual) machines, install a Mahiru registry on one of them and
+some sites on others, and run some kind of distributed processing scenario.
+Mahiru has a set-up for this, and this page explains how to work with it.
+
+Installing software
+-------------------
+
+The distributed set-up uses `Vagrant <https://vagrantup.com>`_ and `Ansible
+<https://ansible.com>`_ to orchestrate the experiment, and we'll also need
+`Docker <https://docker.com>`_ to build some container images. I'm running this
+on Ubuntu Linux, so that's what these instructions are for. If you have a
+different OS, then you'll probably have to do some figuring-out to get things to
+work.
+
+Vagrant creates virtual machines according to a desired configuration and takes
+care of setting up access via SSH. Ansible configures virtual or physical
+machines by connecting to them, examining their state, and taking action to
+ensure it matches some desired state described in its configuration. Vagrant can
+call Ansible to configure (or *provision*) the VMs it has created. Docker is
+used both to deploy Mahiru on the machines and by Mahiru itself to execute
+workflow steps.
+
+If you use Vagrant, then the experiment will be set up locally on a set of
+virtual machines. If you have virtual or physical machines available already,
+then you can skip Vagrant and instead tell Ansible about them, then use Ansible
+directly to install the software and run the experiment.
```

Comment:
```suggestion
We recommend using Vagrant, then the experiment will be set up locally on a set of
virtual machines. For advanced users, if you have virtual or physical machines available already,
then you can skip Vagrant and instead tell Ansible about them, then use Ansible
directly to install the software and run the experiment.
```

Reply:
Same here. I'm actually not explaining how to do this with bare Ansible, without Vagrant. I may try this on the UvA physical machines at some point, so I'll figure that out then and add it.